src/lambda_function/s3_event_config/app.py

```diff
@@ -48,14 +49,61 @@ def lambda_handler(event, context):
         raise KeyError(err_msg)
 
 
-class ExistingNotificationConfiguration(typing.NamedTuple):
-    existing_bucket_notification_configuration: dict
-    existing_notification_configurations_for_type: list
+class NotificationConfigurationType(Enum):
+    """
+    Supported types for an S3 event configuration.
+    """
+    Topic = "Topic"
+    Queue = "Queue"
+    LambdaFunction = "LambdaFunction"
+
+
+class NotificationConfiguration:
+    """
+    An abstraction of S3 event configurations.
+    """
+    def __init__(self, notification_type: NotificationConfigurationType, value: dict):
+        self.type = notification_type.value
+        self.value = value
+        self.arn = self.get_arn()
+
+    def get_arn(self):
+        """
+        Assign the ARN of this notification configuration to the `arn` property.
+        """
+        if self.type == "Topic":
```

Comment:
nit: Since you have an enum you could do:
`if self.type == NotificationConfigurationType.Topic.value`

To prevent repeating the string

Reply:
Initially, this seemed like a more complicated way of achieving the same result. But I like knowing that an IDE would raise a warning on this line if we ever removed support for one of these types.